src/kubeobjects/labels.go

```diff
@@ -21,6 +22,69 @@ const (
 
 type ComponentLabelValue string
 
+type MatchLabels struct {
+	AppName      string
+	AppCreatedBy string
+	AppComponent ComponentLabelValue
+}
+
+type PodLabels struct {
+	MatchLabels
+	AppVersion       string
+	ComponentFeature string
+	ComponentVersion string
+}
+
+func NewMatchLabels(createdBy string, component ComponentLabelValue) *MatchLabels {
+	return &MatchLabels{
+		AppName:      version.AppName,
+		AppCreatedBy: createdBy,
+		AppComponent: component,
+	}
+}
+
+func NewComponentLabels(createdBy string, component ComponentLabelValue, componentFeature string, componentVersion string) *PodLabels {
+	return &PodLabels{
+		MatchLabels:      *NewMatchLabels(createdBy, component),
+		AppVersion:       version.Version,
+		ComponentFeature: componentFeature,
+		ComponentVersion: componentVersion,
+	}
+}
+
+func NewPodLabels(createdBy string, component ComponentLabelValue) *PodLabels {
+	return NewComponentLabels(createdBy, component, "", "")
+}
+
+// BuildLabels produces a set of labels that
+// include versions of operator and component
+// and component feature, if set
```

Comment:
```suggestion
// BuildLabels produces a set of labels that
// contain the matchLabels and 
// extra labels that can change overtime
// or depending the features that are enabled.
```

Nitpick: I think the original comment is a bit too specific, if I want to see the exact labels created then I will look at the code.

The important thing that needs to be mentioned IMO is how its different from `BuildMatchLabels`, which is that it contains labels that can change.
By not being super specific, we will not have to change the comment when we add a new label.